File: Stack/Medium/1441_Build_Array_With_Stack_Operations/Solution.cpp
#include <vector>
#include <string>


class Solution {
public:
    std::vector<std::string> buildArray(std::vector<int>& target, int n) {
        std::vector<std::string> res;

        for (int i{1}, j{}; i <= n && j < target.size(); ++i) {
            res.push_back("Push");
            if (i != target[j]) {
                res.push_back("Pop");
            }
            else {
                ++j;
            }
        }

        return res;
    }
};

File: Stack/Medium/1441_Build_Array_With_Stack_Operations/Solution.py
from typing import List


class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        res = []
        j = 0

        for i in range(1, n + 1):
            res.append("Push")
            if i != target[j]:
                res.append("Pop")
            else:
                j += 1

            if j == len(target):
                break

        return res